### DISCUSSION TITLE
[RFC] Removing `&dyn Node` from `Visit`

### DISCUSSION THREAD

**Discussion by kdy1** [swc-project/swc#2978] [Ideas]
The `&dyn Node` of the methods of `Visit` is useless and annoying. It's actually misdesigned, mainly because `swc` internally used a different visitor system when I made `swc_ecma_visit`.

I want to remove it but I'm not sure if it's used.
I'm going to create an alternative `Visit` if someone needs the `&dyn Node` argument.

**Comment by devongovett** [comment]
I just looked in Parcel's source code and we never use that argument. Sounds good to me!

**Comment by kwonoj** [comment]
I was curious where / when to use it too. +1 for public surface doesn't expose it.
